core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -364,6 +364,12 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   /** Does the configuration contain a given parameter? */
   def contains(key: String): Boolean = settings.containsKey(key)
 
+  /** Returns a copy of this configuration with all the properties at `spark.srcNamespace.xxx`
+    * copied to `spark.xxx` */
+  def fromNamespace(namespace: Namespace): SparkConf = {
```

Comment:
`Namespace` is `private[spark]` but this method is public. Scala should have printed a warning about this. But that means this method should be `private[spark]`.